### PR TITLE
Metadata tool

### PR DESCRIPTION
**Pull request by thinknoack**
a lot of file restructuring - moved download into datasets and files referenced in dataset-routes into the routes directory

### CODE REVIEW

File: packages/openneuro-app/src/scripts/refactor_2021/dataset/mutations/metadata-form.jsx [hunk 1]
```diff
@@ -0,0 +1,394 @@
...
+      additionalProps: {
+        required: false,
+      },
+    },
+    {
+      key: 'trialCount',
+      label: 'Number of Trials (if applicable)',
+      component: NumberInput,
+      additionalProps: {
+        min: -1,
```
Comment (david-nishi) [inline packages/openneuro-app/src/scripts/refactor_2021/dataset/mutations/metadata-form.jsx:84]: Not 0?

File: packages/openneuro-app/src/scripts/refactor_2021/dataset/mutations/metadata-form.jsx [hunk 2]
```diff
@@ -0,0 +1,394 @@
...
+              transformValue,
+            },
+            i,
+          ) => (
+            // @ts-expect-error
+            <FieldComponent
+              name={key}
+              label={label}
+              hoverText={hoverText}
+              value={transformValue ? transformValue(values[key]) : values[key]}
```
Comment (david-nishi) [inline packages/openneuro-app/src/scripts/refactor_2021/dataset/mutations/metadata-form.jsx]:
```suggestion
              value={transformValue ? transformValue(values?.[key]) : values?.[key]}
```

**Review by david-nishi — CHANGES_REQUESTED** [review]

File: packages/openneuro-app/src/scripts/refactor_2021/dataset/mutations/metadata-form.jsx (hunk 1 above)
Comment (thinknoack) [reply packages/openneuro-app/src/scripts/refactor_2021/dataset/mutations/metadata-form.jsx:84]: I just copied this over, not sure why it was -1?

**Review by david-nishi — APPROVED** [review]

**Comment by ckrountree** [comment]
Addresses Gitlab 344

**Pull request merged**